CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 ### Features
 
 ### Bug fixes
+* Changed `BiologicalSequences.distance` to raise an error any time two sequences are passes of different lengths regardless of the `distance_fn` being passed.
```

Comment:
Can you also include a link to the issue number? i.e.:

* your changelog message ([#514](https://github.com/biocore/scikit-bio/issues/514))